src/YoutubeAtom.tsx

```diff
@@ -64,8 +72,172 @@ type YoutubeAtomType = {
+    ophanEventEmitter?: (event: VideoEventKey) => void;
+    getCurrentTime: () => number;
+    getDuration: () => number;
+}) => {
+    /** YouTube API
+            -1 (unstarted)
+            0 (ended)
+            1 (playing)
+            2 (paused)
+            3 (buffering)
+            5 (video cued)
+        **/
+    switch (e.data) {
+        // playing
+        case 1: {
+            // need to make sure user is not resuming
+            if (!hasUserLaunchedPlay) {
+                gaEventEmitter && gaEventEmitter('play');
+                ophanEventEmitter && ophanEventEmitter('play');
+            }
+            setHasUserLaunchedPlay(true);
+
+            progressTracker = setInterval(() => {
+                if (!gaEventEmitter)
+                    // eslint-disable-next-line no-console
+                    console.error(`GA function is not available`);
+                if (!ophanEventEmitter)
+                    // eslint-disable-next-line no-console
+                    console.error(`Ophan function is not available`);
+
+                const currentTime = getCurrentTime();
+                const duration = getDuration();
+                const percentPlayed = Math.round(
+                    (currentTime / duration) * 100,
+                );
+
+                switch (percentPlayed) {
+                    case 25: {
+                        if (!has25PercentEventBeenDispatched) {
+                            gaEventEmitter && gaEventEmitter('25');
+                            ophanEventEmitter && ophanEventEmitter('25');
+                            has25PercentEventBeenDispatched = true;
+                        }
+                        break;
+                    }
+                    case 50: {
+                        if (!has50PercentEventBeenDispatched) {
+                            gaEventEmitter && gaEventEmitter('50');
+                            ophanEventEmitter && ophanEventEmitter('50');
+                            has50PercentEventBeenDispatched = true;
+                        }
+                        break;
+                    }
+                    case 75: {
+                        if (!has75PercentEventBeenDispatched) {
+                            gaEventEmitter && gaEventEmitter('75');
+                            ophanEventEmitter && ophanEventEmitter('75');
+                            has75PercentEventBeenDispatched = true;
+                        }
+                        break;
+                    }
+                }
+            }, 1000);
+            break;
+        }
+        // paused
+        case 2: {
+            progressTracker && clearInterval(progressTracker);
+            break;
+        }
+        // ended
+        case 0: {
+            gaEventEmitter && gaEventEmitter('end');
+            ophanEventEmitter && ophanEventEmitter('end');
+            progressTracker && clearInterval(progressTracker);
+        }
+    }
 };
 
+const overlayStyles = css`
```

Comment:
This CSS doesn't seem related to the PR? Should it be a separate PR?

Reply:
I had to lift up the overlay logic to the parent component. @oliverlloyd then advised to remove the need for the overlay component and to lift it as well. So it not technically part of this PR